sdk/src/test/scala/com/horizen/SidechainStateTest.scala

```diff
@@ -119,7 +120,8 @@ class SidechainStateTest
 
   def getKeyRotationTransaction(boxesWithSecretToOpen: (ZenBox,PrivateKey25519), typeOfKey: KeyRotationProofTypes.KeyRotationProofType, keyIndex: Int, newKeySecret: SchnorrSecret, oldSigningKeySecret: SchnorrSecret, oldMasterKeySecret: SchnorrSecret, wrongNewKey: Boolean = false): CertificateKeyRotationTransaction = {
     val from: JPair[ZenBox,PrivateKey25519] =  new JPair[ZenBox,PrivateKey25519](boxesWithSecretToOpen._1, boxesWithSecretToOpen._2)
-    val messageToSign = newKeySecret.publicImage().getHash
+    val messageToSign = CryptoLibProvider.thresholdSignatureCircuitWithKeyRotation
+      .getMsgToSignForSigningKeyUpdate(newKeySecret.publicImage().pubKeyBytes(), 0, params.sidechainId)
```

Comment:
1) take into account `typeOfKey` same as int the State. Use proper method getMsgToSignForSigningKeyUpdate or getMsgToSignForMasterKeyUpdate same as in the State.
2) add two more arguments to the `getKeyRotationTransaction` method:
* `withdrawalEpochNumber`. Use it here. The passed value must be equal to the one one in the test. Likely, now it is `0`.
* `params: NetworkParams` - at the moment you use `params` field of the test class, but instead you should consider the `mockedParams` local variable of the exact test method, where you need to define `sidechainId` that is used both here and in the `state.validate` later to avoid the nullpointerexception error in the test